packages/workflow/src/utils.ts

```diff
@@ -72,12 +72,38 @@ type MutuallyExclusive<T, U> =
 	| (T & { [k in Exclude<keyof U, keyof T>]?: never })
 	| (U & { [k in Exclude<keyof T, keyof U>]?: never });
 
-type JSONParseOptions<T> = MutuallyExclusive<{ errorMessage: string }, { fallbackValue: T }>;
+type JSONParseOptions<T> = { recovery?: boolean } & MutuallyExclusive<
+	{ errorMessage?: string },
+	{ fallbackValue?: T }
+>;
 
+/**
+ * Parses a JSON string into an object with optional error handling and recovery mechanisms.
+ *
+ * @param {string} jsonString - The JSON string to parse.
+ * @param {Object} [options] - Optional settings for parsing the JSON string. Either `fallbackValue` or `errorMessage` can be set, but not both.
+ * @param {boolean} [options.recovery=false] - If true, attempts to recover from common JSON format errors by replacing single quotes with double quotes and trimming trailing commas.
+ * @param {string} [options.errorMessage] - A custom error message to throw if the JSON string cannot be parsed.
+ * @param {*} [options.fallbackValue] - A fallback value to return if the JSON string cannot be parsed.
+ * @returns {Object} - The parsed object, or the fallback value if parsing fails and `fallbackValue` is set.
+ */
 export const jsonParse = <T>(jsonString: string, options?: JSONParseOptions<T>): T => {
 	try {
 		return JSON.parse(jsonString) as T;
 	} catch (error) {
+		if (options?.recovery) {
+			try {
+				const jsonStringCleaned = jsonString
+					.replace(/'/g, '"') // Replace single quotes with double quotes
+					.replace(/(['"])?([a-zA-Z0-9_]+)(['"])?:/g, '"$2":') // Wrap keys in double quotes
```

Comment:
Do you mind if I have a stab at implementing this with esprima? I think it will be more robust than regex

Reply:
I have not worked with this library and am not sure if it is suited for this. We want to provide some recovery mechanism for common JSON formatting errors, like single quotes instead of double quotes. I would rather avoid bringing in additional dependencies, so it's up to you.

Reply:
We already have it as a dependency (we're using it for our own expressions). I pushed my approach with esprima, let me know what you think. It can handle strings with escaped quotes etc.